### DISCUSSION TITLE
Is it possible to set the width of a DataTable column?

### DISCUSSION THREAD

**Discussion by bjesus** [Textualize/textual#4770] [Q&A] [answered]
I have a 2 columns table. When the terminal screen is narrow, only one column is displayed because the first column contains some rows that can get pretty long. I want the two columns to use the space equally (50/50), and preferably each also have a maximum width that matches the content. This way if the screen is narrow I'll see the beginning of each of the columns, and if the screen is wide the gaps between the columns wouldn't be too big.

Is it possible to do that? I'm not sure which CSS selectors I can play around with, if at all.

Thank you very much!

**Comment by TomJGooding** [comment, marked as answer]
The cells/rows/columns in a `DataTable` aren't widgets, so I'm afraid you won't find any solution using Textual's CSS.

The [`DataTable.add_column`](https://textual.textualize.io/widgets/data_table/#textual.widgets.DataTable.add_column) method does have a `width` param, but that is limited to an absolute value in cells.

Perhaps you could handle the [`Resize`](https://textual.textualize.io/events/resize/) event to dynamically change the column widths, but it might be nice if the `DataTable` provided some easier way of handling this!

**Reply by bjesus** [reply]
Thanks! I ended up using a forked version of [textual-autocomplete](https://github.com/darrenburns/textual-autocomplete) instead.
